### PR TITLE
CORE-6624 Autocomplete react component

### PR DESCRIPTION
**Pull request by homer**

### CODE REVIEW

**Comment by homer** [comment]
`react.socrata.bundle.autocomplete.js file is automatically done by webpack and is there only to have a functioning example file under `examples>example.html`

File: react/autocomplete/src/react.optionTemplate.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    return (
+      <div className={ optionStyleName }>
+        { this.props.data.text }
+      </div>
+    );
+  }
+
+};
+
+OptionTemplate.propTypes = {
```
Comment (rantler) [inline react/autocomplete/src/react.optionTemplate.js]: :+1: :clap:

File: react/autocomplete/src/react.socrata.typeahead.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+          onSelect = { this.handleSelect.bind(this) }
+          onBlur = { this.handleBlur.bind(this) }
+        />
+        <i className={ clearButtonClass } onClick={ this.handleClearSuggestion.bind(this) }></i>
+      </div>
+    )
+  }
+}
+
+SocrataTypeahead.propTypes = {
```
Comment (rantler) [inline react/autocomplete/src/react.socrata.typeahead.js]: Sweet!

**Comment by rantler** [comment]
Looking good! :shipit:

**Pull request merged**